### PR TITLE
Move validation to runtime

### PR DESCRIPTION
**Pull request by daschult**

### CODE REVIEW

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
  ### Size decreased by 5.37% (-107.83 KB)

File: unittests/DSL/TSBuilderTests.cs [hunk 1]
```diff
@@ -180,5 +180,36 @@ public void InsertNewLine()
             builder.InsertNewLine(0);
             Assert.Equal("\n", builder.ToString());
         }
+
+        [Fact]
+        public void AddPropertyToParentObjectWhileBuildingChildObject()
+        {
+            TSBuilder builder = new TSBuilder();
+            Exception exception = null;
```
Comment (RikkiGibson) [inline unittests/DSL/TSBuilderTests.cs]: Consider using the `Exception exception = Assert.Throws(() => builder.Object(...))` pattern

Comment (daschult) [reply unittests/DSL/TSBuilderTests.cs]: I didn't know that Assert.Throws() returned the validated exception. I'll change it to that.

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
  ### Size decreased by 5.37% (-107.83 KB)

**Pull request merged**